### PR TITLE
Modifying Job/Build to store step details

### PR DESCRIPTION
**Pull request by stjohnjohnson**
- Modified Pipeline to write the complete list of permutations
- Modified Build to keep the list of Steps

### CODE REVIEW

File: lib/buildFactory.js [hunk 1]
```diff
@@ -127,9 +127,15 @@ class BuildFactory extends BaseFactory {
 
                 return getCommitSha({ job, scmPlugin: this.scmPlugin, modelConfig })
                     .then(sha => {
-                        modelConfig.sha = sha;
                         // TODO: support matrix jobs
-                        modelConfig.container = job.containers[0];
+                        const index = number.toString().split('.')[1] || 0;
```
Comment (nkatzman) [inline lib/buildFactory.js:127]: when will number ever have a "." in it?

File: lib/log.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     * @param  {String}   config.message        Contents of the line
+     *
+     */
+    constructor(config) {
+        super('log', config);
+    }
+
+    /**
+     * Lazy load a build model
+     * @property pipeline
```
Comment (nkatzman) [inline lib/log.js]: build

File: lib/log.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+     * @return {Promise}    Resolves to the build associated with this build
```
Comment (nkatzman) [inline lib/log.js]: Hey I got this build for your build

File: lib/buildFactory.js (hunk 1 above)
Comment (stjohnjohnson) [reply lib/buildFactory.js:127]: it will never in this PR.  Until we solve line 130

**Comment by nkatzman** [comment]
👍

**Comment by d2lam** [comment]
Looks fine 👍

**Pull request merged**